optimizely/config/polling_manager.go

```diff
@@ -69,16 +69,20 @@ func (cm *PollingProjectConfigManager) SyncConfig(datafile []byte) {
 	}
 
 	projectConfig, err := datafileprojectconfig.NewDatafileProjectConfig(datafile)
+
+	cm.configLock.Lock()
 	if err != nil {
 		cmLogger.Error("failed to create project config", err)
-	}
+	} else {
+		var previousRevision = "null"
```

Comment:
This is not right. You are trying to set a zero value string right? You should just declare the variable 'var previousRevision string'

Reply:
actually not, just following java implementation, it prints "null" in case of initial datafile assignment, it can be seen here:
https://github.com/optimizely/java-sdk/blob/c3e55cec05a18dc2761ae7c64d98933efdbbcbf4/core-api/src/main/java/com/optimizely/ab/config/PollingProjectConfigManager.java#L93